### ISSUE TITLE
redundant termios 

### ISSUE THREAD

**Issue by sktwj** [membermatters/pymultidropbus#1]
why not modify its parity directly?

ser.parity = serial.PARITY_EVEN
ser.apply_settings()

it seems termios can be removed. yes?

**Comment by jabelone** [comment]
Hey there, sorry I didn't get a notification for your issue. That will not work because that is not how the MDB protocol works.

The MDB protocol is a 9 bit serial protocol whereas your typical serial devices use 8 bits. We're (mis)using the normally unused "parity bit" feature of a UART which happens to be in the same place a 9th bit would be. Using some software trickery we can detect when this 9th "parity" bit is a 1 or 0, and force it to a 1 or 0 when needed when sending bytes. Reading and writing the state of this 9th bit is an important part of the MDB protocol and required to differentiate between address bytes.

Just setting the serial port to `serial.PARITY_EVEN` won't work because MDB doesn't use a parity bit as explained above. There's also no way to reliably change this setting "on the fly" using pyserial (give it a try, it doesn't work). The second issue is you need to *detect* if the 9th bit is set or not. If the serial byte fails the parity bit test it is usually discarded so we need to tell the system not to discard it, but mark it instead.

To recap, termios is necessary for the following reasons:
1. Dynamically set the parity bit to mark (always 1) or space (always 0) "on the fly" for each byte we send.
2. Tell the system not to discard bytes that fail the parity bit and instead mark them (Linux does this by prepending 0xFF 0x00).
